core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -398,27 +398,29 @@ public static void mergeIndexAndWriteSegmentFile(CarbonTable carbonTable, String
    * @throws IOException
    */
   public static String writeSegmentFile(CarbonTable carbonTable, String segmentId, String UUID,
```

Comment:
for the update, it will have more than one timstamp, right

Reply:
Agree, Let me check and modify

Reply:
I have pushed now. waiting for the build

Reply:
@QiangCai : I have thought about, only non update scenario I can handle this issue. For update there is no easy way currently to find out which is stale and which is not. One way for update is to read old segment file and add files that timestamp greater than old segment file content + old segment file content.